src/components/PracticeInterface.tsx

```diff
@@ -111,191 +176,162 @@ const PracticeInterface: React.FC<PracticeInterfaceProps> = ({
-                  totalScore: newTotalScore,
-                  bestScore: newBestScore,
-                };
-              });
-            }
           },
           {
-            duration: 60000, // 1 minute practice session
-            onComplete: () => {
-              setPracticeState("completed");
-              setIsTracking(false);
-              const finalScore = session.bestScore;
-              onComplete?.(finalScore);
-            },
+            duration: Infinity, // Run until we manually stop
             onError: (error) => {
-              setError(error.message);
-              setIsTracking(false);
-            },
+              console.error("Tracking error", error);
+            }
           }
         );
-
         stopTrackingRef.current = stopTracking;
+
       } else if (activity.type === "movement" && activity.movementData) {
-        // Start movement sequence comparison
+        // Original logic for movement
+        setPracticeState("practicing");
+        setIsTracking(true);
+
         const stopTracking = await mediaPipeService.startMovementTracking(
           videoRef.current,
-          (landmarks, timestamp) => {
+          (landmarks) => {
             setCurrentLandmarks(landmarks);
           },
           {
             duration: activity.duration || 10000,
             onComplete: async () => {
-              try {
-                // Compare recorded movement with target movement
-                const recordedSequence: TimestampedLandmarks[] = [];
-                // This would be populated during tracking - simplified for now
-
-                const result = await comparisonService.compareMovementSequence(
-                  recordedSequence,
-                  activity.movementData!,
-                  difficulty
-                );
-
-                setComparisonResult(result);
-                setPracticeState("completed");
-                setIsTracking(false);
-
-                // Update final session stats
-                setSession((prev) => ({
-                  ...prev,
-                  attempts: prev.attempts + 1,
-                  successfulMatches: result.isMatch
-                    ? prev.successfulMatches + 1
-                    : prev.successfulMatches,
-                  totalScore: prev.totalScore + result.score,
-                  bestScore: Math.max(prev.bestScore, result.score),
-                }));
-
-                onComplete?.(result.score);
-              } catch (error) {
-                const message =
-                  error instanceof Error
-                    ? error.message
-                    : "Failed to compare movement";
-                setError(message);
-                setIsTracking(false);
-              }
+              stopPractice();
+              setPracticeState("completed");
             },
             onError: (error) => {
-              setError(error.message);
+              handleError(error.message);
```

Comment:
in the movement onError callback you call handleError(error.message) and setIsTracking(false) but don’t invoke stopPractice(), so a failing tracking session might leave the camera running until something else shuts it down. It would be safer to also stop tracking/stream there